File: src/com/leetcode/Easy/MaximumNestingDepthOfTheParentheses.java
package com.leetcode.Easy;

import java.util.*;

// 1614. Maximum Nesting Depth of the Parentheses
public class MaximumNestingDepthOfTheParentheses {
    public int maxDepth(String s) {
        int maxDepth = 0;
        Stack<Character> stack = new Stack<>();
        for (char ch : s.toCharArray()) {
            if (ch == '(')
                stack.push(ch);

            if (ch == ')') {
                if (stack.peek() == '(') {
                    maxDepth = Math.max(stack.size(), maxDepth);
                    stack.pop();
                }
            }
        }
        return maxDepth;
    }
}

File: JavaScript/Easy/MaximumNestingDepthOfTheParentheses.js
/**
 * 1614. Maximum Nesting Depth of the Parentheses
 * @param {string} s
 * @return {number}
 */
const maxDepth = function (s) {
  let maxDepth = 0;
  const stack = [];
  for (let i = 0; i < s.length; i++) {
    const ch = s[i];
    if (ch === '(') {
      stack.push(ch);
    }
    if (ch === ')') {
      if (stack.length > 0 && stack[stack.length - 1] === '(') {
        maxDepth = Math.max(stack.length, maxDepth);
        stack.pop();
      }
    }
  }
  return maxDepth;
};